engine/access/rpc/backend/backend_transactions.go

```diff
@@ -782,8 +802,8 @@ func (b *backendTransactions) getTransactionResultFromExecutionNode(
 	return events, resp.GetStatusCode(), resp.GetErrorMessage(), nil
 }
 
-func (b *backendTransactions) NotifyFinalizedBlockHeight(height uint64) {
-	b.retry.Retry(height)
+func (b *backendTransactions) ProcessFinalizedBlockHeight(height uint64) error {
```

Comment:
:warning: 
This might be ok for now but this might be a source of problems in future. We run this code on finalization gorotuine, potentially lagging finalization events if operations take long time. We might need to move this logic on dedicated goroutine and provide a way to skip finalization events if they are delivered too often for this engine. 
Here is an [example](https://github.com/onflow/flow-go/blob/master/engine/common/follower/compliance_engine.go#L201) of similar approach.

I think we should at least write a comment about it and please describe return types and expected errors if there are any(I suspect that not)